src/dune_pkg/outdated_packages.mli

```diff
@@ -0,0 +1,50 @@
+open Import
+
+(** [t] represents the result of searching for outdated packages in a lock directory. *)
+type t
+
+(** [find ~repos ~local_packages lock_dir] searches for outdated packages in the given
+    [lock_dir] by consulting the [repos] and [local_packages].*)
+val find
+  :  repos:Opam_repo.t list
+  -> local_packages:OpamFile.OPAM.t Package_name.Map.t
+  -> Lock_dir.t
+  -> t
+
+(** [pp ~transitive ~lock_dir_path t] returns a specially constructed user message
+    explaining the outdated packages. It begins with a summary detailing the number of
+    affected packages and then lists all the packages requested.
+
+    - [transitive] indiciates whether to hint that transitive dependencies are not being
+      shown and therefore the user should pass [--transitive] to the [dune pkg outdated]
+      command to see them.
+
+    - [lock_dir_path] is the path to the lock directory that will appear in the messages. *)
+val pp
+  :  transitive:bool
+  -> lock_dir_path:Path.Source.t
+  -> t
+  -> User_message.Style.t Pp.t list
+
+(** [raise_errors ~raise ~lock_dir_path t] raises an errors that occured whilst searching
+    for outdated packages. This allows the errors to be delayed so that other results can be
+    printed first. *)
+val raise_errors
+  :  default:User_message.Style.t Pp.t list
+  -> t
+  -> User_message.Style.t Pp.t list
```

Comment:
Why does have a function that raises an error have a non polymorphic return type?

Reply:
I've mixed up some concerns here. It is printing spaces when there are no errors.